src/Executables/ExportCoordinates/ExportCoordinates.hpp

```diff
@@ -110,8 +114,148 @@ struct ExportCoordinates {
+            typename ArrayIndex>
+  static void apply(db::DataBox<DbTags>& /*box*/,
+                    const Parallel::CBase_GlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const double& overall_min_grid_spacing) noexcept {
+    printf("Overall inertial minimum grid spacing: %f\n\n",
+           overall_min_grid_spacing);
+  }
+};
 }  // namespace Actions
 
+template <size_t Dim, typename Metavars>
+struct SingletonParallelComponent {
+  using chare_type = Parallel::Algorithms::Singleton;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using metavariables = Metavars;
+  using phase_dependent_action_list = tmpl::list<Parallel::PhaseActions<
+      typename Metavars::Phase, Metavars::Phase::Initialization, tmpl::list<>>>;
+  using initialization_tags = Parallel::get_initialization_tags<
+      Parallel::get_initialization_actions_list<phase_dependent_action_list>>;
+
+  static void execute_next_phase(
+      const typename Metavars::Phase /*next_phase*/,
+      const Parallel::CProxy_GlobalCache<Metavars>& /*global_cache*/) {}
+};
+
+template <typename Metavars, typename PhaseDepActionList,
+          class ImportInitialData = ImportNoInitialData>
+struct ElementArray;
+
+template <size_t Dim, typename PhaseDepActionList, class ImportInitialData>
+struct ArrayReduce {
+  template <typename ParallelComponent, typename DbTags, typename Metavars,
+            typename ArrayIndex>
+  static void apply(const db::DataBox<DbTags>& box,
+                    const Parallel::GlobalCache<Metavars>& cache,
+                    const ArrayIndex& array_index) noexcept {
+    if constexpr (db::tag_is_retrievable_v<
+                      domain::Tags::MinimumGridSpacing<Dim, Frame::Inertial>,
+                      db::DataBox<DbTags>>) {
+      static_assert(
+          std::is_same_v<
+              ParallelComponent,
+              ElementArray<Metavars, PhaseDepActionList, ImportInitialData>>,
+          "The ParallelComponent is not deduced to be the right type");
+
+      const auto& my_proxy = Parallel::get_parallel_component<
+          ElementArray<Metavars, PhaseDepActionList, ImportInitialData>>(
+          cache)[array_index];
+      const auto& singleton_proxy = Parallel::get_parallel_component<
+          SingletonParallelComponent<Dim, Metavars>>(cache);
+
+      Parallel::ReductionData<Parallel::ReductionDatum<double, funcl::Min<>>>
+          elemental_min_grid_spacing{
+              get<domain::Tags::MinimumGridSpacing<Dim, Frame::Inertial>>(box)};
+      Parallel::contribute_to_reduction<Actions::PrintMinimumGridSpacing>(
+          elemental_min_grid_spacing, my_proxy, singleton_proxy);
+
+    } else {
+      ERROR(
+          "You're missing the domain::Tags::MinimumGridSpacing<Dim, "
+          "Frame::Inertial> in the DataBox.");
+    }
+  }
+};
+
+template <typename Metavars, typename PhaseDepActionList,
+          class ImportInitialData>
+struct ElementArray
```

Comment:
I don't think you need this parallel component. Just stick with the `DgElementArray`, convert the `ArrayReduce` action into an iterable action (you only need to change the `apply` function signature, look at `Parallel/Actions/TerminatePhase.hpp` for an example) and add it to one of the phase action lists below (e.g. the `Export` phase). You don't even need the extra phase you added. If any of this is not clear, I'm happy to explain more, just ping me on Slack or here :)

While you're at it, please rename `ArrayReduce` to something more descriptive, perhaps `FindGlobalMinimumGridSpacing` (or something better, if you have an idea).